src/cellular/dataCollection.cpp

```diff
@@ -0,0 +1,123 @@
+
+    LocationPoint point;
+    LocationService& instance= LocationService::instance();
+
+    
+    bool hasGPS = false;
+    Ensemble10_eventData_t* pData = &ensemble10Data;
+
+    #pragma pack(push, 1)
+    struct
+    {
+        EnsembleHeader_t header;
+        union{
+            Ensemble10_data_t ens10;
+            Ensemble11_data_t ens11;
+        }data;
+    }ensData;
+    #pragma pack(pop)
+
+    getAccelerometer(accelData, accelData + 1, accelData + 2);
+    getGyroscope(gyroData, gyroData + 1, gyroData + 2);
+    getMagnetometer(magData, magData + 1, magData + 2);
+
+    /**
+     * Check pre-processer defines to switch between 
+     * ICM and MAX temp sensor
+     */
+    #if USE_ICM_TEMP_SENSOR
+        getTemperature(&temp);
+    #else 
+        temp = pSystemDesc->pTempSensor->getTemp();
+    #endif
+        
+    instance.getLocation(point);
+
+    if(point.locked == 1 && point.satsInView > 4)
+    {
+        hasGPS = true;
+        lat = point.latitude;
+        lng = point.longitude;
+    }
+    else
+    {
+        hasGPS = false;
+        lat = pData->location[0];
+        lng = pData->location[1];
+    }
+
+    water = 0; // TODO - waiting on water sensor integration
+
+    // Accumulate measurements
+    pData->temperature += temp;
+    pData->water += water;
+    pData->acc[0] += B_TO_N_ENDIAN_2(accelData[0]);
+    pData->acc[1] += B_TO_N_ENDIAN_2(accelData[1]);
+    pData->acc[2] += B_TO_N_ENDIAN_2(accelData[2]);
+    pData->ang[0] += B_TO_N_ENDIAN_2(gyroData[0]);
+    pData->ang[1] += B_TO_N_ENDIAN_2(gyroData[1]);
+    pData->ang[2] += B_TO_N_ENDIAN_2(gyroData[2]);
+    pData->mag[0] += B_TO_N_ENDIAN_2(magData[0]);
+    pData->mag[1] += B_TO_N_ENDIAN_2(magData[1]);
+    pData->mag[2] += B_TO_N_ENDIAN_2(magData[2]);
+    pData->location[0] += lat;
+    pData->location[1] += lng;
+    pData->hasGPS += hasGPS ? 1 : 0;
+    pData->accumulateCount++;
+
+    // ensData.header.elapsedTime_ds = Ens_getStartTime(pDeployment->startTime);
+    // SF_OSAL_printf("Ensemble timestamp: %d\n", ensData.header.elapsedTime_ds);
+    ensData.data.ens10.rawTemp = N_TO_B_ENDIAN_2(temp / 0.0078125);
+    ensData.data.ens10.rawAcceleration[0] = N_TO_B_ENDIAN_2(pData->acc[0])  ;
+    ensData.data.ens10.rawAcceleration[1] = N_TO_B_ENDIAN_2(pData->acc[1])  ;
+    ensData.data.ens10.rawAcceleration[2] = N_TO_B_ENDIAN_2(pData->acc[2])  ;
+    ensData.data.ens10.rawAngularVel[0] = N_TO_B_ENDIAN_2(pData->ang[0])  ;
+    ensData.data.ens10.rawAngularVel[1] = N_TO_B_ENDIAN_2(pData->ang[1])  ;
+    ensData.data.ens10.rawAngularVel[2] = N_TO_B_ENDIAN_2(pData->ang[2])  ;
+    ensData.data.ens10.rawMagField[0] = N_TO_B_ENDIAN_2(pData->mag[0])  ;
+    ensData.data.ens10.rawMagField[1] = N_TO_B_ENDIAN_2(pData->mag[1])  ;
+    ensData.data.ens10.rawMagField[2] = N_TO_B_ENDIAN_2(pData->mag[2])  ;
+    ensData.data.ens11.location[0] = N_TO_B_ENDIAN_4(pData->location[0])  ;
+    ensData.data.ens11.location[1] = N_TO_B_ENDIAN_4(pData->location[1])  ;
```

Comment:
If there is accumulation happening, shouldn't there then be division at some point to get mean values? And shouldn't this only happen once everything is completely accumulated?